phoenix-core/src/main/java/org/apache/phoenix/parse/ParseNodeRewriter.java

```diff
@@ -157,6 +152,14 @@ public static SelectStatement rewrite(SelectStatement statement, ParseNodeRewrit
             }
             normGroupByNodes.add(normGroupByNode);
         }
+
+        ParseNode having = statement.getHaving();
```

Comment:
Do this is just an ordering problem? Looks good to me.

Reply:
Thanks @lhofhansl . Yes, just an ordering problem. After adjusting the order, we can resolve alias in having clause.